### DISCUSSION TITLE
Running TOP locally: Is recommended setting of rbenv to 3.1.2 still necessary?

### DISCUSSION THREAD

**Discussion by Breadpaw** [TheOdinProject/theodinproject#4226] [General]
When installing TOP locally, I ran into the following comment in the instructions:

**Note: If bundle install doesn't work and you get a "rbenv version" error of some kind, try running the following command to set your Ruby version in the project.**

~~~
rbenv local 3.1.2
~~~

[Link to the page and section](https://github.com/TheOdinProject/theodinproject/wiki/Running-The-Odin-Project-Locally#install-gems-and-migrate-the-database)

My question: The [TOP installation guide](https://www.theodinproject.com/lessons/ruby-installing-ruby) currently instructs to install and globally set Ruby version 3.2.2. Does this make this comment unnecessary?

**Comment by KevinMulhern** [comment]
Nice catch! thanks @Breadpaw 💪 I've updated the version. You have a point about it being unnecessary. It _might_ be good to keep around as a guard rail for anyone that has Ruby installed and goes straight into the setting up locally guide. But I need to think on that more.

**Comment by Breadpaw** [comment]
Thanks @KevinMulhern , also for the quick reply :)

I considered suggesting a comment in [this section](https://github.com/TheOdinProject/theodinproject/wiki/Running-The-Odin-Project-Locally#initial-installations) for people to consider updating Ruby via the [Installing Ruby](https://www.theodinproject.com/lessons/ruby-installing-ruby) lesson for their respective system. Just a highlight, so that long-time or returning students are reminded of their possible earlier installs of Ruby (as was the case with my system)

**Comment by KevinMulhern** [comment]
Thinking about this a bit more, the Ruby version used on the site and on the curriculum don't always match. When a new version of Ruby comes out, we update the curriculum fairly quickly. But theres usually a lag time between the new versions release and the dependencies we use on the site getting updates to support it. From how thats played out in the past, it could be anywhere from a few months to a year before we can safely update and be running the site on the same version used in the curriculum.

I think we should continue linking to the install guide lesson, it's a good general guide. But have a new step in the [Install Gems and Migrate the Database](https://github.com/TheOdinProject/theodinproject/wiki/Running-The-Odin-Project-Locally#install-gems-and-migrate-the-database) section advising contributors to check they have the correct Ruby version installed. The source of truth for that will always be in our [.ruby-version](https://github.com/TheOdinProject/theodinproject/blob/main/.ruby-version) file. 

Thanks again, for raising this @Breadpaw. This has probably caught more than a few people out when we've been between versions in the past.
